src/main/java/org/testng/internal/MethodHelper.java

```diff
@@ -461,16 +454,14 @@ public static void dumpInvokedMethodsInfoToConsole(
   }
 
   protected static String calculateMethodCanonicalName(Class<?> methodClass, String methodName) {
+    StringBuilder result = new StringBuilder();
     Set<Method> methods = ClassHelper.getAvailableMethods(methodClass); // TESTNG-139
-    Method result = null;
-    for (Method m : methods) {
-      if (methodName.equals(m.getName())) {
-        result = m;
-        break;
-      }
-    }
+    methods.stream()

```

Comment:
Try

```java
return methods.stream()
        .filter(method -> methodName.equals(method.getName()))
        .findFirst()
        .ifPresent(method -> calculateMethodCanonicalName(method)) // calculateMethodCanonicalName can return null
        .orElse(null);
```

Reply:
`ifPresent()` and `orElse()` I don't think work together, because `ifPresent()` returns `null`

Reply:
Right!

```java
return methods.stream()
        .filter(method -> methodName.equals(method.getName()))
        .findFirst()
        .flatMap(method -> calculateMethodCanonicalName(method)) // calculateMethodCanonicalName can return null
        .orElse(null);
```

Reply:
Fixed.